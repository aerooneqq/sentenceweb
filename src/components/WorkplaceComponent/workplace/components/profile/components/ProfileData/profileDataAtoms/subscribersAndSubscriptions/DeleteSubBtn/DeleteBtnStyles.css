.deleteSubBtn{ 
    width: 80px;
    height: 30px;

    border-width: 1px;
    border-color: red;
    border-style: solid;
    border-radius: 7px;

    font-family: Arial;
    color: white;
    font-size: 13px;

    background: red;
}

.deleteSubBtn:hover{ 
    background: linear-gradient(-120deg, #ee0000, red);
}

.deleteSubBtn:focus{ 
    outline: none;
}