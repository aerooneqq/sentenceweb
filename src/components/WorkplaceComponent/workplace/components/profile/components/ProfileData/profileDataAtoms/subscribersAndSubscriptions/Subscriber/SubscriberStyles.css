.friendsListContainer{ 
    width: 90%;
    background: transparent;

    padding-top: 10px;
    padding-bottom: 10px;

    display: flex;
    flex-direction: row;

    border-style: solid;
    border-width: 0px;
    border-color: #d0d0d0;
    border-bottom-width: 0.8px;
    border-radius: 2px;
}

.friendsListContainer:nth-last-child(1){
    border-width: 0; 
}


.friendsListUserPhoto{ 
    width: 50px;
    height: 50px;

    border-radius: 50px;
    border-color: orangered;
    border-width: 1px;
    border-style: solid;
}

.friendsListUserInfo{ 
    flex-grow: 1;
    background: transparent;

    display: flex;
    flex-direction: column;
}

.friendsListUserName{ 
    margin-left: 15px;

    font-family: Arial;
    font-size: 17px;
    color: #505050;

    justify-self: baseline;
}

.friendsListUserSeparatorCont{
    flex-grow: 1;
}

.friendsListUserBirthDate{ 
    margin-left: 15px;

    font-family: Arial;
    color: gray;
    font-size: 11px;
}

.friendsListUserBtns{ 
    margin-left: 15px;
    margin-top: 10px;
}