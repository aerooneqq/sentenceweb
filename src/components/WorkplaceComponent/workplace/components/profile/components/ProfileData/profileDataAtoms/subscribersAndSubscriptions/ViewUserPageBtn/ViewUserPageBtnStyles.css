.viewUserPageBtn{ 
    width: 80px;
    height: 30px;

    background: gray;

    border-style: solid;
    border-radius: 7px;
    border-color: gray;
    border-width: 1px;
    
    margin-left: 10px;

    font-family: Arial;
    color: white;
}

.viewUserPageBtn:focus{ 
    outline: none;
}

.viewUserPageBtn:hover{ 
    background: linear-gradient(-120deg,#707070, #808080)
}