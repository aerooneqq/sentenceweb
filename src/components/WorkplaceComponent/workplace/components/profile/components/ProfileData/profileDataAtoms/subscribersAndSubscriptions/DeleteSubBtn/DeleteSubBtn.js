import React, {Component} from "react";

//Styles
import "./DeleteBtnStyles.css";

/**
 * PROPS LIST:
 * 1) deleteSubscriber
 * 2) userID
 */
export default class DeleteSubBtn extends Component { 
    constructor(props) { 
        super(props);
        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick() { 
        this.props.deleteSubscriber(this.props.userID);
    }

    render() { 
        return( 
            <button className = "deleteSubBtn"
                    onClick = {this.handleBtnClick}>
                Delete
            </button>
        )
    }
}