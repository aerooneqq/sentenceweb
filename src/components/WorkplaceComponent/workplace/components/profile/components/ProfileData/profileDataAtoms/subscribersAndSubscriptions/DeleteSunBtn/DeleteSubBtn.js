import React, {Component} from "react";

//Styles
import "./DeleteBtnStyles.css";

export default class DeleteSubBtn extends Component{ 
    constructor(props){ 
        super(props);
        this.handleBtnClick = this.handleBtnClick.bind(this);

        alert(this.props.userID);
    }

    handleBtnClick(){ 
        this.props.deleteSubscriber(this.props.userID);
    }

    render(){ 
        return( 
            <button class = "deleteSubBtn"
                    onClick = {this.handleBtnClick}>
                Delete
            </button>
        )
    }
}