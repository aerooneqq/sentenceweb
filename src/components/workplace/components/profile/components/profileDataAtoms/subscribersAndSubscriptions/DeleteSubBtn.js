import React, {Component} from "react";

//Styles
import "./styles/DeleteBtnStyles.css";

export default class DeleteSubBtn extends Component{ 
    constructor(props){ 
        super(props);
    }

    render(){ 
        return( 
            <button class = "deleteSubBtn">
                Delete
            </button>
        )
    }
}