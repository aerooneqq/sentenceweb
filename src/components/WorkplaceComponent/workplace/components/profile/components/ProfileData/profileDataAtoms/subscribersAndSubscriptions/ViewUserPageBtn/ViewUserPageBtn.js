import React, {Component} from "react";

//Styles
import "./ViewUserPageBtnStyles.css"

/**
 * PROPS LIST: 
 * 1) viewUserPage
 * 2) userID
 */
export default class ViewUserPageBtn extends Component { 

    constructor(props) { 
        super(props);

        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick() { 
        this.props.viewUserPage(this.props.userID);
    }

    render() { 
        return( 
            <button className = "viewUserPageBtn"
                    onClick = {this.handleBtnClick}>
                View
            </button>
        )
    }
}