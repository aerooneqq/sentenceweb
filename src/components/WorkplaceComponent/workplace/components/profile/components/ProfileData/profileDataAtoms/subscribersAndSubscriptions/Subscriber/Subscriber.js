import React, {Component, lazy} from "react";

//Styles
import "./SubscriberStyles.css";

//Components
import DeleteSubBtn from "../DeleteSubBtn/DeleteSubBtn";
import ViewUserPageBtn from "../ViewUserPageBtn/ViewUserPageBtn";

/**
 * PROPS LIST:
 * 1) subscriber (birthDate, name, userID),
 * 2) deleteSubscriber
 */
export default class Subscriber extends Component {  
    render() {  
        let date = this.props.subscriber.birthDate;
        date = date.substr(0, 4) + "-" + date.substr(5, 2) + "-" + date.substr(8, 2);

        return (
                <div className = "friendsListContainer">
                    <div className = "friendsListUserPhoto">
                    </div>
    
                    <div className = "friendsListUserInfo">
                        <div className = "friendsListUserName">
                            {this.props.subscriber.name}
                        </div>
                        <div className = "friendsListUserBirthDate">
                            {date}
                        </div>
                        <div class = "friendsListUserBtns">
                            <DeleteSubBtn deleteSubscriber = {this.props.deleteSubscriber}
                                          userID = {this.props.subscriber.userID}/>
                            <ViewUserPageBtn />
                        </div>
                    </div>
                </div>
        );
    }
}