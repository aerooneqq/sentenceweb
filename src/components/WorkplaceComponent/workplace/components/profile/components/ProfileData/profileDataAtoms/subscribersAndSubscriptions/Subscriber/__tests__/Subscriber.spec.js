import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import Subscriber from "../Subscriber";

import DeleteSubBtn from "../../DeleteSubBtn/DeleteSubBtn";
import ViewUserPageBtn from "../../ViewUserPageBtn/ViewUserPageBtn";

const propsSpy = jest.fn(() => {});
const props = { 
    subscriber: { 
        name: "Aero", 
        userID: 1,
        birthDate: "30.05.2000T23131212"
    }, 
    deleteSubscriber: propsSpy 
}

describe("Subscriber", () => {
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<Subscriber {...props}/>))).toMatchSnapshot();
    });

    it("should display right data", () => { 
        const subscriber = shallow(<Subscriber {...props}/>);

        expect(subscriber.find("div.friendsListUserName").text()).toBe(props.subscriber.name);

        const {subscriber: {birthDate}} = props;
        let rightDate =  birthDate.substr(0, 4) + "-" + birthDate.substr(5, 2) + 
            "-" + birthDate.substr(8, 2);
        expect(subscriber.find("div.friendsListUserBirthDate").text()).toBe(rightDate);
    });

    it("should contain btns", () => { 
        const subscriber = shallow(<Subscriber {...props}/>);

        expect(subscriber.find(DeleteSubBtn).exists()).toBe(true);
        expect(subscriber.find(ViewUserPageBtn).exists()).toBe(true);
    })
})