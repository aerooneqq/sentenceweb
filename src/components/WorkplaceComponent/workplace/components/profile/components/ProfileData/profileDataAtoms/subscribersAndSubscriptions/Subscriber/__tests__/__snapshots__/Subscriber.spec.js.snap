// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscriber should render correctly 1`] = `
<div
  className="friendsListContainer"
>
  <div
    className="friendsListUserPhoto"
  />
  <div
    className="friendsListUserInfo"
  >
    <div
      className="friendsListUserName"
    >
      Aero
    </div>
    <div
      className="friendsListUserBirthDate"
    >
      30.0-.2-00
    </div>
    <div
      class="friendsListUserBtns"
    >
      <DeleteSubBtn
        deleteSubscriber={[MockFunction]}
        userID={1}
      />
      <ViewUserPageBtn />
    </div>
  </div>
</div>
`;
